lib/ansible/plugins/action/net_platform_agnostic.py

```diff
@@ -0,0 +1,191 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import sys
+import copy
+
+from ansible.plugins.action import ActionBase
+from ansible.utils.path import unfrackpath
+from ansible.plugins import connection_loader
+from ansible.module_utils.basic import AnsibleFallbackNotFound
+from ansible.module_utils.six import iteritems
+from ansible.module_utils._text import to_bytes
+
+from imp import find_module, load_module
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        if self._play_context.connection != 'local':
+            return dict(
+                failed=True,
+                msg='invalid connection specified, expected connection=local, '
+                    'got %s' % self._play_context.connection
+            )
+
+
+        play_context = copy.deepcopy(self._play_context)
+        play_context.network_os = play_context.network_os or self._get_network_os(task_vars)
+        play_context.connection = 'network_cli' # TODO this can be netconf
+
+        self.provider = self._load_provider(play_context.network_os)
+
+        play_context.remote_addr = self.provider['host'] or self._play_context.remote_addr
+        play_context.port = self.provider['port'] or self._play_context.port or 22
+        play_context.remote_user = self.provider['username'] or self._play_context.connection_user
+        play_context.password = self.provider['password'] or self._play_context.password
+        play_context.private_key_file = self.provider['ssh_keyfile'] or self._play_context.private_key_file
+        play_context.timeout = self.provider['timeout'] or self._play_context.timeout
+        if 'authorize' in self.provider.keys():
+            play_context.become = self.provider['authorize'] or False
+            play_context.become_pass = self.provider['auth_pass']
+
+        socket_path = self._start_connection(play_context)
+        task_vars['ansible_socket'] = socket_path
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+
+        module = self._get_implementation_module(play_context.network_os, self._task.action)
+
+        if not module:
+            result['failed'] = True
+            result['msg'] = 'Could not find implementation module for %s' % play_context.network_os
```

Comment:
```
- result['msg'] = 'Could not find implementation module for %s' % play_context.network_os
+ result['msg'] = 'Could not find implementation module %s for %s' % (self._task.action, play_context.network_os)
```
This would be more verbose:
```
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Could not find implementation module net_system for eos"
}
```